net/pfSense-pkg-Quagga_OSPF/files/usr/local/pkg/quagga_ospfd.inc

```diff
@@ -514,24 +558,86 @@ function quagga_ospfd_plugin_carp($pluginparams) {
 	if (is_array($config['installedpackages']['quaggaospfd']['config'])) {
 		$ospfd_conf = &$config['installedpackages']['quaggaospfd']['config'][0];
 	} else {
+		syslog(LOG_ALERT, "Quagga: CONFIGURATION DOES NOT EXIST!");
```

Comment:
Is that really worth logging an error for? The user may have it installed but not configured yet. Doesn't seem fatal.

Reply:
This error is to alert a user in the event that they have selected to follow a CARP interface for the purposes of OSPF changes, but have not actually created an OSPF configuration. Since the default state is "off" for CARP tracking, we found it useful in our logging to be sure we didn't unnecessarily have it set to follow CARP events.

Reply:
But that will be executed even if the user hasn't chosen to follow a CARP interface. It will trigger for any CARP event where the user has the package installed but not configured. That isn't an error state.

Reply:
I see what you're saying. Sorry, it's been over a year since this was written and I haven't looked at it since then. If I remember correctly, this was some debug logging from making sure all configuration variations produced the expected behavior. We can probably safely remove this.